cmp/viewmanager/DataAccess.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2024 Extremely Heavy Industries Inc.
+ */
+
+import {PlainObject, XH} from '@xh/hoist/core';
+import {pluralize, throwIf} from '@xh/hoist/utils/js';
+import {map} from 'lodash';
+import {ViewInfo} from './ViewInfo';
+import {View} from './View';
+import {ViewManagerModel} from './ViewManagerModel';
+
+export interface ViewCreateSpec {
+    name: string;
+    group: string;
+    description: string;
+    isShared: boolean;
+    value?: PlainObject;
+}
+
+export interface ViewUpdateSpec {
+    name: string;
+    group: string;
+    description: string;
+    isShared?: boolean;
+    isDefaultPinned?: boolean;
+}
+
+export interface ViewUserState {
+    currentView?: string;
+    userPinned: Record<string, boolean>;
+    autoSave: boolean;
+}
+
+/**
+ * Supporting class for accessing and updating ViewManager and View data.
+ *
+ * @internal
+ */
+export class DataAccess<T> {
```

Comment:
A general question - do we have any other classes like this, or named like this?  Just trying to relate this to any patterns we already have for breaking up functionality along these lines.

Reply:
I couldn't find a direct analog, and as you might have noticed, I have been playing around with various names for it.

The class gets a lot thinner with the move to a server-side view service, so probably less necessary but thought it was worth it to keep it around, with the main class being so large.